### PR TITLE
Terraform resource for MongoDB Custom Roles

### PR DESCRIPTION
**Pull request by PacoDw**
**Work in progress**. Opening a draft pull request to show progress on this and perhaps occasionally ask for advice.

Adding a new resource for managing Custom MongoDB Roles. This follows my earlier pull request mongodb/go-client-mongodb-atlas#20 adding the necessary functionality to the Go client.

### Resource
- [x] Schema
- [x] Creation
- [x] Reading
- [x] Updating
- [x] Deletion
- [x] Import
- [x] Testing file
- [x] Documentation
### Data source
- [x] Schema
- [x] Reading
- [x] Testing file
- [x] Documentation

### CODE REVIEW

**Comment by PacoDw** [comment]
@mpaluchowski We can continue since this pr :) I was able to recover the previous work 👍

**Comment by PacoDw** [comment]
Hi @mpaluchowski! Good news everything is done, now we need to wait until the review to be able to merge it to the master branch. Thank you so much for your contribution and all your effort, well done!

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"role_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`[\w\d-]+`), "`role_name` can contain only letters, digits, underscores, and dashes"),
```
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_custom_db_role.go:38]: Nice catch!

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+					func(v interface{}, k string) (ws []string, es []error) {
+						value := v.(string)
+						log.Printf("LOG___ value: %#+v\n", value)
```
Comment (marinsalinas) [inline mongodbatlas/resource_mongodbatlas_custom_db_role.go]: Could you please remove this log?

**Review by marinsalinas — CHANGES_REQUESTED** [review]
LGTM, just remove logs and fix the conflicts, (SDK v0.1.3 is released you are able to use it now)

File: mongodbatlas/resource_mongodbatlas_custom_db_role.go (hunk 2 above)
Comment (PacoDw) [reply mongodbatlas/resource_mongodbatlas_custom_db_role.go]: Nice! thank you so much :) let me remove it 👍

**Comment by themantissa** [comment]
@PacoDw  is this ready for a MDB review?

**Comment by PacoDw** [comment]
yes @themantissa, it is, goes ahead! 👍

File: website/docs/r/custom_db_role.html.markdown [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+sidebar_current: "docs-mongodbatlas-resource-custom-db-role"
+description: |-
+    Provides a Custom DB Role resource.
+---
+
+# mongodbatlas_custom_db_role
+
+`mongodbatlas_custom_db_role` provides a Custom DB Role resource. The customDBRoles resource lets you retrieve, create and modify the custom MongoDB roles in your cluster. Use custom MongoDB roles to specify custom sets of actions which cannot be described by the built-in Atlas database user privileges.
+
+-> **IMPORTATN** Custom roles cannot use actions unavailable to any cluster version in your project. Custom roles are defined at the project level, and must be compatible with each MongoDB version used by your project’s clusters. If you have a cluster in your project with MongoDB 3.4, you cannot create a custom role that uses actions introduced in MongoDB 3.6, such as useUUID.
```
Comment (themantissa) [inline website/docs/r/custom_db_role.html.markdown]: Misspelled IMPORTANT

File: website/docs/r/custom_db_role.html.markdown [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+	* Is a name already used by an existing custom role in the project
+	* Is a name of any of the built-in roles
+	* Is `atlasAdmin`
+	* Starts with `xgen-`
+
+
+### Actions
+Each object in the actions array represents an individual privilege action granted by the role. It is an required field.
+
+* `action` - (Required) Name of the privilege action. For a complete list of actions available in the Atlas API, see Custom Role Actions.
```
Comment (themantissa) [inline website/docs/r/custom_db_role.html.markdown]: Add the link and a bit more detail:
Name of the privilege action. For a complete list of actions available in the Atlas API, see 
[Custom Role Actions] (https://docs.atlas.mongodb.com/reference/api/custom-role-actions).
Note: The privilege actions available to the Custom Roles API resource represent a subset of the privilege actions available in the Atlas Custom Roles UI.

File: website/docs/r/custom_db_role.html.markdown [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+```
+
+## Argument Reference
+
+* `project_id` - (Required) The unique ID for the project to create the database user.
+* `role_name` - (Required) Name of the custom role. 
+
+	-> **IMPORTATN** The specified role name can only contain letters, digits, underscores, and dashes. Additionally, you cannot specify a role name which meets any of the following criteria:
```
Comment (themantissa) [inline website/docs/r/custom_db_role.html.markdown]: Misspelling: IMPORTANT

**Review by themantissa — CHANGES_REQUESTED** [review]
A few small edits to the docs but otherwise LGTM!

File: website/docs/r/custom_db_role.html.markdown (hunk 3 above)
Comment (PacoDw) [reply website/docs/r/custom_db_role.html.markdown]: goog catch! thanks :)

**Review by marinsalinas — APPROVED** [review]
LGTM!

**Review by themantissa — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**